ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Block/SupportsSanityCheck.hs

```diff
@@ -0,0 +1,84 @@
+-- | This module adds support for sanity checking consensus configuration
+--   on node startup. These checks should primarily look for unusual
+--   configuration choices that may point to an accidentally-misconfigured node
+--   and quietly cause problems, rather than incoherent configurations that will
+--   result in fatal errors at a later point.
+--
+--   While in most situations they can be handled as fatal issues, there are
+--   situations when intentionally using "weird" configuration can be useful,
+--   and so the user should be able to opt out of the sanity checks at their
+--   own peril.
+module Ouroboros.Consensus.Block.SupportsSanityCheck (
+    BlockSupportsSanityCheck (..)
+  , SanityCheckIssue (..)
+  , checkSecurityParamConsistency
+  , issueDescription
+  , sanityCheckConfig
+  ) where
+
+import           Control.Exception
+import           Data.List.NonEmpty (NonEmpty (..))
+import qualified Data.List.NonEmpty as NonEmpty
+import           Data.Maybe (catMaybes)
+import           Ouroboros.Consensus.Config (TopLevelConfig)
+import           Ouroboros.Consensus.Config.SecurityParam
+
+-- | An issue found in the 'TopLevelConfig' for a block. See 'issueDescription'
+--   for human-readable descriptions of each of these cases, especially when
+--   presenting these to users.
+data SanityCheckIssue
+  -- | Configuration contains multiple security parameters. This may cause
+  --   strange behaviour around era boundaries.
+  = InconsistentSecurityParam (NonEmpty SecurityParam)
+  deriving (Show, Eq)
+
+instance Exception SanityCheckIssue
+
+issueDescription :: SanityCheckIssue -> String
+issueDescription issue = case issue of
+  InconsistentSecurityParam ks ->
+    mconcat
+      [ "SecurityParams (K) were found to be inconsistent between constituent "
+      , "eras of a HardForkBlock: "
+      , show (NonEmpty.toList ks)
+      ]
+
+-- | "BlockSupportsSanityCheck" provides evidence that a block can be sanity
+--   checked for common issues on node startup. "sanityCheckConfig", which runs
+--   performs each check and returns a list with each "SanityCheckIssue" found,
+--   should be preferred over using these methods directly.
+class BlockSupportsSanityCheck blk where
+
+  -- | Generate a "NonEmpty" list of security parameters for a given block type.
+  --   For individual eras' block types, this is simply a singleton list
+  --   containing the chosen "SecurityParam", but combined block types will
```

Comment:
Please explicitly list the HardForkCombinator as an example of a "combined block type".